### PR TITLE
The real AME nerf

### PR DESCRIPTION
**Pull request by EmoGarbage404**
## About the PR

does a few things to AME

- rebalances the power curve to give more power at low core counts but less at higher core count
- cap the fuel injection to coreCount x 8
- rebalance core damage to scale with excess injection amount
- halve the amount of fuel per AME canister

## Why / Balance

image

the AME is meant to be a bootstrapping / emergency power supply. It shouldn't be able to power full-sized stations completely for 30+ minutes.

## Media

image
power curves

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl:
- tweak: Changed AME power output. Lower injection amounts now produce more power but further injections have diminishing returns.
- tweak: Increased damage per injection overclocked AMEs.
- tweak: Halved fuel per AME jar.

### CODE REVIEW

**Comment by Partmedia** [comment]
What's being shown in the plot? It looks like it's power output vs either injection or core count?

**Comment by EmoGarbage404** [comment]
> What's being shown in the plot? It looks like it's power output vs either injection or core count?

power output vs injection at 1 core

**Comment by CyberDuck-Git** [comment]
Finally, he's getting a nerfed

**Comment by LankLTE** [comment]
Is the AME still going to be able to power lowpop stations reliably with this change?

**Comment by Brandon-Huu** [comment]
Bro did not just do their math in desmos

**Comment by EmoGarbage404** [comment]
> Is the AME still going to be able to power lowpop stations reliably with this change?

afaict most lowpop stations use a 4-core AME so they should be able to power it just fine. They'll probably need more fuel mapped if they want to soley rely on it but it shouldn't be a huge issue.

**Comment by quatre** [comment]
Will 120kW leave enough time to setup other power sources before SMES drain? From what I remember, that looks about half what the station uses and would leave something like 8 minutes to spin-up other sources.

File: Content.Server/Ame/AmeNodeGroup.cs [hunk 1]
```diff
@@ -134,22 +134,11 @@ public float InjectFuel(int fuel, out bool overloading)
...
-        // overloadVsSizeResult > 10: This will explode in at most 20 injections.
-        if (overloadVsSizeResult > 10)
-            instability = 5;
-
-        // Apply calculated instability
-        if (instability == 0)
-            return powerOutput;
+        var instability = overloadVsSizeResult / CoreCount;
+        var fuzz = _random.Next(-1, 1);
+        instability += fuzz; // fuzz the values a tiny bit.
```
Comment (metalgearsloth) [inline Content.Server/Ame/AmeNodeGroup.cs:141]: This is going to be -1, 0 which I assume is what you want then?

**Review by metalgearsloth — APPROVED** [review]

File: Content.Server/Ame/AmeNodeGroup.cs (hunk 1 above)
Comment (EmoGarbage404) [reply Content.Server/Ame/AmeNodeGroup.cs]: i fucked it up but i corrected it

**Comment by tsp01** [comment]
seems this
fixes: #29423

**Comment by Golinth** [comment]
Does this mean that AME parts could theoretically be added back to cargo?

I loved the silliness of making absolutely massive AMEs for no reason, and with diminishing returns it could be balanced again

**Comment by marc-pelletier** [comment]
It makes absolutely no sense that a 4-core AME with an optimal injection rate makes less power than a 2 core AME with an optimal injection rate. 2 core AME with 4 injections produces 180kW, whereas a 4 core AME with 8 injections produces 152kW? Why was this even merged?

**Comment by osjarw** [comment]
> 2 core AME with 4 injections produces 180kW, whereas a 4 core AME with 8 injections produces 152kW?

Tested this, this is indeed what's happening.

**Comment by thebadman4662** [comment]
> It makes absolutely no sense that a 4-core AME with an optimal injection rate makes less power than a 2 core AME with an optimal injection rate. 2 core AME with 4 injections produces 180kW, whereas a 4 core AME with 8 injections produces 152kW? Why was this even merged?

I am playing on hopper so AME nerf did not affect me yes and I might be clueless but isnt...

obraz

...this basically saying you should use 8 injections at 2 core and 16 with 4?

**Comment by osjarw** [comment]
> ...this basically saying you should use 8 injections at 2 core and 16 with 4?

I haven't taken a close look at the code yet, testing with 4 cores and 16 injection produces 203.2kW, and is causing CoreIntegrity to go down, I also think it caused AME overloading adminlogs.

I think I have this PR, since these fuel jars contain just 500 fuel

**Comment by YotaXP** [comment]
Assuming I entered the formulae correctly, this curve shows it punishing you for using more than 2 cores.
Blue = Original, Red = New
https://www.desmos.com/calculator/ovi6ef8roa
image

**Pull request merged**